### ISSUE TITLE
Vulnerability Detector: adapt the NVD scan for macOS agents

### ISSUE THREAD

**Issue by chemamartinez** [wazuh/wazuh#6085]
### Description

During the research at https://github.com/wazuh/wazuh/issues/5881 we have detected some details that have to adapt during the NVD scan for macOS agents.

- For macOS agents, the CPEs of Operating Systems have to be included and reported to detect vulnerabilities like https://nvd.nist.gov/vuln/detail/CVE-2020-9859.

- The field `target_sw` have to be taken into account when looking for vulnerabilities affecting to macOS software:

Here you can see the interesting variants of this field.

```
sqlite> select count(*),target_sw from nvd_cpe where target_sw like "mac%" group by target_sw;
20|mac
12|mac_os
54|mac_os_x
188|macos
```

- The kernel package is not needed to be scanned for macOS agents, kernel vulnerabilities are handled with OS patches.

See:
https://support.apple.com/en-gb/HT210119
https://nvd.nist.gov/vuln/detail/CVE-2019-8591

- Packages with no version should be skipped

- Packages with no vendor should be analyzed but ignoring the vendor

**Comment by danimegar** [comment]
Development branch: https://github.com/wazuh/wazuh/compare/5860-mac-vdt...6085-adapt-nvd-scan-macos

**Comment by chemamartinez** [comment]
Merged #6290
